crates/datasources/src/postgres/mod.rs

```diff
@@ -642,6 +647,54 @@ impl TableProvider for PostgresTableProvider {
 
         Ok(Arc::new(exec))
     }
+
+    async fn insert_into(
+        &self,
+        state: &SessionState,
+        input: Arc<dyn ExecutionPlan>,
+        _overwrite: bool,
+    ) -> DatafusionResult<Arc<dyn ExecutionPlan>> {
```

Comment:
I think the implementation works for a first pass. I do think there's some opportunities to avoid a single insert (where we collect everything in memory).

- We may be able to do copy in (https://docs.rs/tokio-postgres/latest/tokio_postgres/struct.Transaction.html#method.copy_in) and write to a buffer.
- We could have an alternative execution plan implementation that streams the output of `input` into postgres without the intermediate batch writing.

Reply:
Yup, that makes sense! I will create an issue for this.

Reply:
#1888